dockstore-webservice/src/test/java/io/dockstore/webservice/helpers/ZenodoHelperTest.java

```diff
@@ -172,11 +180,54 @@ void testDefaultDoiOrderOfPrecedence() {
     }
 
+    @Test
+    void testIdFromZenodoDoi() {
+        assertEquals("5104875", ZenodoHelper.idFromZenodoDoi("10.5281/zenodo.5104875"));
+        assertNull(ZenodoHelper.idFromZenodoDoi(null));
+        assertEquals("", ZenodoHelper.idFromZenodoDoi("notavaliddoi"));
+    }
+
+    @Test
+    void testFindGitHubIntegrationDois() throws JsonProcessingException {
+        final SearchResult searchResult = deserialize(FixtureUtility.fixture("fixtures/zenodoListRecords.json"));
+        final List<ZenodoHelper.ConceptAndDoi> dois = ZenodoHelper.findGitHubIntegrationDois(searchResult.getHits().getHits(), "coverbeck/cwlviewer");
+        assertEquals(List.of(new ZenodoHelper.ConceptAndDoi("10.5281/zenodo.11094520", "10.5281/zenodo.11099749")), dois);
+    }
+
+
+    @Test
+    void testTaggedVersions() throws JsonProcessingException {
+        final SearchResult searchResult = deserialize(FixtureUtility.fixture("fixtures/zenodoVersions.json"));
+        final List<TagAndDoi> taggedVersions = ZenodoHelper.findTaggedVersions(searchResult.getHits().getHits(), "coverbeck/cwlviewer");
+        final List<TagAndDoi> expected = List.of(
+                new TagAndDoi("FourthTestTag", "10.5281/zenodo.11099749"),
+                new TagAndDoi("ThirdTestTag", "10.5281/zenodo.11095575"),
+                new TagAndDoi("SecondTestTag", "10.5281/zenodo.11095507"),
+                new TagAndDoi("testtag", "10.5281/zenodo.11094521"));
+        assertEquals(expected, taggedVersions);
+    }
+
+    @Test
+    void testTagFromRelatedIdentifier() {
+        assertEquals(Optional.empty(), ZenodoHelper.tagFromRelatedIdentifier("dockstore/dockstore-cli", "https://github.com/dockstore/dockstore-cli/mytag"));
+        assertEquals(Optional.empty(), ZenodoHelper.tagFromRelatedIdentifier("dockstore/dockstore-cli", "nonsense-icalstring"));
+        assertEquals("mytag", ZenodoHelper.tagFromRelatedIdentifier("dockstore/dockstore-cli", "https://github.com/dockstore/dockstore-cli/tree/mytag").get());
+        assertEquals("1.16", ZenodoHelper.tagFromRelatedIdentifier("dockstore/dockstore-cli", "https://github.com/dockstore/dockstore-cli/tree/1.16").get());
+    }
+
+
     private DockstoreWebserviceConfiguration createDockstoreConfiguration() {
         final DockstoreWebserviceConfiguration config = new DockstoreWebserviceConfiguration();
         config.getExternalConfig().setBasePath("/api/");
         config.getExternalConfig().setHostname("dockstore.org");
         config.getExternalConfig().setScheme("https");
         return config;
     }
+    private static SearchResult deserialize(String fixture) throws JsonProcessingException {
+        final ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        final SearchResult searchResult = objectMapper.readValue(fixture, SearchResult.class);
+        return searchResult;
+    }
+
```

Comment:
This feels like something we might already be doing somewhere else in the codebase, I wonder if there's a helper (or if we should create one).